Core/Core/Router/Router.swift

```diff
@@ -197,9 +197,9 @@ open class Router {
             view.navigationItem.leftItemsSupplementBackButton = true
         }
 
-        var nav: UINavigationController?
+        var nav: StyledNavigationController?
         if options.embedInNav {
-            nav = view as? UINavigationController ?? UINavigationController(rootViewController: view)
+            nav = view as? StyledNavigationController ?? StyledNavigationController(rootViewController: view)
```

Comment:
The original solution allowed custom `UINavigationController` subclasses to be used as navigation controllers. This modified behaviour only allows `StyledNavigationController` so if I wanted to push a `HelmNavigationController` I couldn't because this router would replace it. I'm not sure if this would cause any harm as I think we wouldn't call this method with `embedInNav: true` and a nav controller already in the stack, so this is just something to think about.